src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java

```diff
@@ -174,4 +177,69 @@ private int getFeeds() throws IOException {
     urlConn.disconnect();
     return responseCode;
   }
+
+  @Test
+  public void testHTTPSinkMacroUrl() throws Exception {
+    String inputDatasetName = "input-http-sink-with-macro";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+    Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+      .put("url", "${url}")
```

Comment:
please add a negative test when url is null or empty

Reply:
PR updated.